pkg/sql/sqlinstance/instanceprovider/instanceprovider.go

```diff
@@ -35,7 +35,42 @@ type writer interface {
 	ReleaseInstanceID(ctx context.Context, instanceID base.SQLInstanceID) error
 }
 
-// provider implements the sqlinstance.Provider interface for access to the sqlinstance subsystem.
+// placeholderProvider implements the sqlinstance.Provider interface as a
+// placeholder for an instance provider, when an instance provider must be
+// instantiated for a non-SQL instance. It starts a Reader to provide the
+// AddressResolver interface, but otherwise throws unsupported errors.
+type placeholderProvider struct {
+	*instancestorage.Reader
+}
+
+func NewPlaceholder() sqlinstance.Provider {
+	return &placeholderProvider{}
+}
+
+// Start implements the sqlinstance.Provider interface.
+func (p *placeholderProvider) Start(ctx context.Context) error {
+	return nil
+}
+
+// Instance implements the sqlinstance.Provider interface.
+func (p *placeholderProvider) Instance(
+	ctx context.Context,
+) (_ base.SQLInstanceID, _ sqlliveness.SessionID, err error) {
+	return base.SQLInstanceID(0), "", sqlinstance.NotASQLInstanceError
+}
+
+// GetInstance implements the AddressResolver interface.
+func (p *placeholderProvider) GetInstance(context.Context, base.SQLInstanceID) (sqlinstance.InstanceInfo, error) {
+	return sqlinstance.InstanceInfo{}, sqlinstance.NotASQLInstanceError
+}
+
+// GetAllInstances implements the AddressResolver interface.
+func (p *placeholderProvider) GetAllInstances(context.Context) ([]sqlinstance.InstanceInfo, error) {
+	return nil, sqlinstance.NotASQLInstanceError
+}
```

Comment:
I can see just defining this in the `sqlinstance` package. It doesn't know anything about implementation details that this package knows about.